src/com/google/enterprise/gsafeed/GsafeedHelper.java

```diff
@@ -88,34 +92,77 @@ public void fatalError(SAXParseException exception)
           throw exception;
         }
       });
-    Document document = docBuilder.parse(inputStream);
-    return (Gsafeed) unmarshaller.unmarshal(document);
+    JAXBContext jaxbContext =
+        JAXBContext.newInstance("com.google.enterprise.gsafeed");
```

Comment:
The String parameter is the directory that has the JAXB classes?   

How are non JAXB classes in this package treated? 

Is there another way to specify the classes so that compiler checks that package is the one holding the JAXB classes?


Is there a way to make this work where the JAXB classes are innner static classes?

Reply:
The string is the package name. One or more package names can be specified, with the restriction that there can't be name conflicts in the XML element names. The package must have one or both of: an ObjectFactory class, or a file named jaxb.index containing the class names of the JAXB classes. JAXBContext has alternative newInstance method that lets you specify the classes explicitly; it will automatically pick up classes referenced by a class, so just specifying the root element class can be enough. (Javadoc for javax.xml.bind.JAXBContext)

Non-JAXB classes seem to be ignored - ours aren't causing a problem.

I think that using jaxb.index or, possibly, extra annotations on the ObjectFactory methods, would let you specify inner classes as the representations of XML elements. I haven't tried either so far.

Reply:
(1) Should we use reflection, like this.getClass().getPackage().toString() ? or something like so? I'm not sure. I want to consider alternative where compiler is more involved in detecting error.  Perhaps one of the JAXB classes can have static method that returns an unmarshaller?

(2) What are your current thoughts on putting JAXB classes in seperate package or in wrapping separte class as inner classes?

Reply:
Using getPackage() seems good; several examples I came across do that. I'll look at that next.

Since we were starting with the unmodified generated classes, I used the helper classes to make it easier to write the examples. I thought that maybe as we start modifying the JAXB classes we might choose to let Gsafeed or something own the marshal/unmarshal process. 

I don't have a strong preference about how to structure the JAXB classes, assuming it all works. We may end up with some more classes as we proceed to better handle data (content adapters, marshal/unmarshal listeners, that sort of thing).